examples/h5_ref_compat.c

```diff
@@ -76,9 +76,9 @@ main(void)
 
     /* Access reference and read dataset data through new API */
     assert(H5Rget_type((const H5R_ref_t *)&new_ref_buf[0]) == H5R_OBJECT2);
-    H5Rget_obj_type3((const H5R_ref_t *)&new_ref_buf[0], H5P_DEFAULT, &obj_type);
+    H5Rget_obj_type3((H5R_ref_t *)&new_ref_buf[0], H5P_DEFAULT, &obj_type);
```

Comment:
Is the cast necessary? Looks like the type of &new_ref_buf[0] should be H5R_ref_t *.

Reply:
Indeed not; will fix.